### PR TITLE
MCP-502; fix: allow underscores in server ID validation

### PR DESCRIPTION
**Pull request by dhanesh-gore-fluidai**
Server IDs like `pdf_to_excel_lic` were rejected due to regex
pattern `[a-z0-9-]` not allowing underscores.

Updated pattern to `[a-z0-9_-]` in:
- Backend: Pydantic model in api.py
- Frontend: ServerForm and CloneFromGithubForm validation,
  error messages, help text, and placeholder

Added 9 unit tests covering valid and invalid server ID patterns.

### CODE REVIEW

File: fluidmcp/cli/models/api.py [hunk 1]
```diff
@@ -247,7 +247,7 @@ class AddServerFromGitHubRequest(BaseModel):
         ),
         min_length=1,
         max_length=100,
-        pattern="^[a-z0-9-]+$",
+        pattern="^[a-z0-9_-]+$",
     )
```
Comment (shubhendu-shekhar-fluidai) [inline fluidmcp/cli/models/api.py:251]: missing pattern — [api.py:29]

id: str = Field(..., description="Unique server identifier", min_length=1, max_length=100)
No pattern= here, unlike at line 250 which has pattern="^[a-z0-9_-]+$".

Should add pattern="^[a-z0-9_-]+$" for consistency.

File: fluidmcp/cli/services/server_builder.py [hunk 2]
```diff
@@ -26,22 +26,22 @@ def slugify(text: str) -> str:
...
-        # Remove non-alphanumeric characters (except hyphens)
-        text = re.sub(r'[^a-z0-9-]', '', text)
+        # Replace whitespace with hyphens (underscores preserved)
+        text = re.sub(r'\s+', '-', text)
+        # Remove non-alphanumeric characters (except hyphens and underscores)
+        text = re.sub(r'[^a-z0-9_-]', '', text)
         # Collapse consecutive hyphens
         text = re.sub(r'-+', '-', text)
-        return text.strip('-')
+        return text.strip('-_')
```
Comment (shubhendu-shekhar-fluidai) [inline fluidmcp/cli/services/server_builder.py:44]: Consecutive underscore collapsing. Something similar for consecutive hypens is already present.
Add text = re.sub(r'_+', '_', text) between these two lines (43-44)

**Comment by shubhendu-shekhar-fluidai** [comment]
Since you have added new tests as well. Consider adding this edge case test as well: 
slugify with consecutive underscores and leading/trailing underscore after stripping

Other changes look ok to me.

**Review by shubhendu-shekhar-fluidai — APPROVED** [review]

**Review by abhinav-fluidai — APPROVED** [review]
## Security Review

**Blast Radius: LOW**
Isolated validation change — backend Pydantic pattern, Python slugify utility, two frontend form components, and unit tests. No auth, no API surface change.

### Summary
Expands the server ID regex from `^[a-z0-9-]+$` to `^[a-z0-9_-]+$` consistently across backend model, slugify util, and both frontend forms. Frontend also guards against IDs starting/ending with underscore (mirrors the existing hyphen check). 9 new unit tests cover valid and invalid patterns against the live Pydantic model. No injection risk — values are validated before use, never interpolated unsafely.

### Issues Found
None.

### Verdict
APPROVE
